### ISSUE TITLE
OptionallyTypedRule is not exported to Lint.Rules namespace

### ISSUE THREAD

**Issue by bgold09** [palantir/tslint#2536]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: N/A
- __Running TSLint via__: N/A

#### TypeScript code being linted

N/A

#### Actual behavior

The following does not compile:

```typescript
import * as Lint from 'tslint';
export class SomeNewRule extends Lint.Rules.OptionallyTypedRule {
    // ...
}
```

The [OptionallyTypedRule](https://github.com/palantir/tslint/blob/master/src/language/rule/optionallyTypedRule.ts) is not being imported with AbtractRule and TypedRule in the Lint.Rules namespace; this is no more than a minor inconvenience when authoring custom rules. 

#### Expected behavior

The code above should compile. The workaround is easy, and is to directly reference ```d.ts``` for OptionallyTypedRule, e.g.:

```typescript
import { OptionallyTypedRule } from "../node_modules/tslint/lib/language/rule/optionallyTypedRule";
export class SomeNewRule extends OptionallyTypedRule {
    // ...
}
```
